src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java

```diff
@@ -112,37 +115,57 @@ protected void setFunctionReferences(List<LfSensitivityFactor> factors) {
-            // solve system
+            assert Integer.MAX_VALUE / (engine.getEquationSystem().getSortedEquationsToSolve().size() * Double.BYTES) > factorGroups.size();
             DenseMatrix factorsStates = initFactorsRhs(lfNetwork, engine.getEquationSystem(), factorGroups); // this is the rhs for the moment
             j.solveTransposed(factorsStates);
             setFunctionReferences(lfFactors);
+            sensitivityValues.addAll(calculateSensitivityValues(factorGroups, factorsStates));
 
-            // calculate sensitivity values
-            sensitivityValues = calculateSensitivityValues(factorGroups, factorsStates);
+            assert Integer.MAX_VALUE / (engine.getEquationSystem().getSortedEquationsToSolve().size() * Double.BYTES) > additionalFactorGroups.size();
+            DenseMatrix additionalFactorsStates = initFactorsRhs(lfNetwork, engine.getEquationSystem(), additionalFactorGroups); // this is the rhs for the moment
+            j.solveTransposed(additionalFactorsStates);
+            setFunctionReferences(additionalLfFactors);
+            sensitivityValues.addAll(calculateSensitivityValues(additionalFactorGroups, additionalFactorsStates));
         }
-
-        LfContingency.reactivateEquations(deactivatedEquations, deactivatedEquationTerms);
         return sensitivityValues;
     }
 
     /**
      * https://people.montefiore.uliege.be/vct/elec0029/lf.pdf / Equation 32 is transposed
      */
-    public Pair<List<SensitivityValue>, Map<String, List<SensitivityValue>>> analyse(Network network, List<SensitivityFactor> factors, List<PropagatedContingency> contingencies,
+    public Pair<List<SensitivityValue>, Map<String, List<SensitivityValue>>> analyse(Network network, SensitivityFactorsProvider factorsProvider, List<PropagatedContingency> contingencies,
                                                                                      LoadFlowParameters lfParameters, OpenLoadFlowParameters lfParametersExt) {
         Objects.requireNonNull(network);
-        Objects.requireNonNull(factors);
         Objects.requireNonNull(lfParameters);
         Objects.requireNonNull(lfParametersExt);
 
         // create LF network (we only manage main connected component)
         List<LfNetwork> lfNetworks = LfNetwork.load(network, new LfNetworkParameters(lfParametersExt.getSlackBusSelector(), false, true, lfParameters.isTwtSplitShuntAdmittance(), false, lfParametersExt.getPlausibleActivePowerLimit(), false));
         LfNetwork lfNetwork = lfNetworks.get(0);
         checkContingencies(lfNetwork, contingencies);
-        checkSensitivities(network, factors);
         checkLoadFlowParameters(lfParameters);
+
+        List<SensitivityFactor> factors = factorsProvider.getCommonFactors(network);
+        checkSensitivities(network, factors);
+        List<SensitivityFactor> additionalFactors = factorsProvider.getAdditionalFactors(network);
```

Comment:
The SensitivityFactoryProvider API is terrible, I add to read to JavaDoc to figure out that this is factors that have to be calculated for only base case.... In our code maybe we should be more clear by naming like factorsOnBaseCase or something like this?